backend/consensus/base.py

```diff
@@ -225,7 +225,7 @@ async def exec_transaction(
             if transaction.leader_only:
                 remaining_validators = []
 
-            num_validators = len(remaining_validators) + 1
+            num_validators = len(remaining_validators) + 1  # +1 for the leader
```

Comment:
The idea of the issue was that we'd use the same leader's node to run as a validator with the rest. That would mean that we should add it to the validator set, as another validator, without any customization